extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -196,9 +196,13 @@ public PanacheQueryType find(Class<?> entityClass, String query, Object... param
     public PanacheQueryType find(Class<?> entityClass, String query, Sort sort, Object... params) {
         String findQuery = PanacheJpaUtil.createFindQuery(entityClass, query, paramCount(params));
         EntityManager em = getEntityManager(entityClass);
-        // FIXME: check for duplicate ORDER BY clause?
         if (PanacheJpaUtil.isNamedQuery(query)) {
             String namedQuery = query.substring(1);
+            if (sort != null) {
+                throw new IllegalArgumentException(
+                        "Sort cannot be used with named query, add an order by clause to the named query " + namedQuery
+                                + " instead");
```

Comment:
That looks better, but I miss some quotes in the error message, maybe also display the entity containing the named query would help?

`You cannot use Sort.by on named queries. Try adding an "order by" clause to the named query "xxx" defined in the entity "yyyy" instead`

Reply:
I think about the quotes but decided it's not worth it but I can add them.
Regarding the entity, I have the information but as a named query is unique it's easily findable.
Keep in mind that the issue will arrise at developement time, at this time the developer knew what it just modified

Reply:
Quotes added!